src/Oxpecker/README.md

```diff
@@ -999,6 +1000,42 @@ Just like in the previous examples the record type must be decorated with the `[
 
 The underlying model binder is configured as a dependency during application startup (see [Binding Forms](#binding-forms))
 
+### Model validation
+
+Oxpecker diverges from the Giraffe's approach to model validation and embraces the traditional ASP.NET Core model validation based on `System.ComponentModel.DataAnnotations` attributes.
+
+While you might still need to do complex validation inside your domain, the built-in DTO model validation is still useful for the API boundary.
+
+You have 3 ways to validate your model:
+- Directly using `validateModel` function
+- Using `ctx.BindAndValidate*` extension methods (similar to `ctx.Bind*`)
+- Using `bindAndValidate*` handlers (similar to `bind*`)
+
+Inside handler you'll need to match `ValidationResult` to handle both valid and invalid cases:
+
+```fsharp
+let addCar : EndpointHandler =
+    fun (ctx: HttpContext) ->
+        task {
+            match! ctx.BindAndValidateJson<Car>() with
+            | ValidationResult.Valid car ->
+                return! ctx.Write <| Ok car
+            | ValidationResult.Invalid (invalidCar, errors) ->
+                return! ctx.Write <| BadRequest errors.All
+        }
+```
+
+If you are using server-side rendering using `Oxpecker.ViewEngine`, you can leverage special  `ModelState`
+```fsharp
+[<RequireQualifiedAccess>]
+type ModelState<'T> =
+    | Empty
```

Comment:
Maybe `NotValidated` to prevent any collision.
However, if you require qualified access this is fine too

Reply:
Yes, `RequireQualifiedAccess` resolves any collisions